struct.go

```diff
@@ -326,8 +341,34 @@ func (s *Section) mapTo(val reflect.Value, isStrict bool) error {
 	return nil
 }
 
-// MapTo maps section to given struct.
-func (s *Section) MapTo(v interface{}) error {
+// maps all sections with the same name and returns the new value.
+func (s *Section) mapAllTo(secName string, val reflect.Value, isStrict bool) (reflect.Value, error) {
+	typ := val.Type()
+	if typ.Kind() == reflect.Slice {
+		if secs, err := s.f.GetSections(secName); err == nil {
+			for _, sec := range secs {
+				sliceType := typ.Elem()
+
+				// create new element
+				newElement := reflect.New(sliceType)
+				if err = sec.mapTo(newElement, isStrict); err != nil {
+					return reflect.Value{}, fmt.Errorf("error mapping section(%s): %v", secName, err)
+				}
+
+				// add element to field
+				val = reflect.Append(val, newElement.Elem())
+			}
+			return val, nil
+		} else {
+			return reflect.Value{}, err
+		}
+	} else {
+		return reflect.Value{}, fmt.Errorf("error mapping section(%s): Val has to be of type Slice", secName)
+	}
```

Comment:
```suggestion
	if typ.Kind() != reflect.Slice {
		return reflect.Value{}, fmt.Errorf("error mapping section(%s): Val has to be of type Slice", secName)
	}
	
	secs, err := s.f.GetSections(secName)
	if err != nil {
		return reflect.Value{}, err
	} 
	
	for _, sec := range secs {
		sliceType := typ.Elem()

		// create new element
		newElement := reflect.New(sliceType)
		if err = sec.mapTo(newElement, isStrict); err != nil {
			return reflect.Value{}, fmt.Errorf("error mapping section(%s): %v", secName, err)
		}

		// add element to field
		val = reflect.Append(val, newElement.Elem())
	}
	return val, nil
```

Reply:
Ah, now I I understand what you mean. Thank you for the suggestion.